cli/pkg/printer/printer.go

```diff
@@ -71,10 +108,64 @@ func (p *Printer) PrintResource(v interface{}) error {
 		return nil
 	case JSON:
 		return p.PrintJSON(v)
+	case CSV:
+		return p.PrintCSV(v)
+	case TSV:
+		return p.PrintTSV(v)
 	}
 	return fmt.Errorf("unknown printer.Format: %T", *p.format)
 }
 
+// PrintCSV prints the given data as CSV to the output.
+func (p *Printer) PrintCSV(v interface{}) error {
+	var out io.Writer = os.Stdout
+	if p.resourceOut != nil {
+		out = p.resourceOut
+	}
+
+	type csvvaluer interface {
+		MarshalCSVValue() interface{}
+	}
+	if c, ok := v.(csvvaluer); ok {
+		v = c.MarshalCSVValue()
+	}
```

Comment:
What is the need for this? It doesn't look like anything in our code implements `MarshalCSVValue`